src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -29,6 +29,52 @@ def setUp(self):
   def tearDown(self):
     pass
 
+  @winnforum_testcase
+  def test_WINNF_FT_S_SIQ_4(self):
+    """Tests related to DPA activated for some channels
+    
+    SAS sending responseCode = 0 successful response
+    and including all channels within frequency range FR.
+    """
+    # Trigger SAS to load DPAs
+    self._sas_admin.TriggerLoadDpas()
+    # Trigger SAS to de-active all the DPAs 
+    self._sas_admin.TriggerDpaActivation({'activate':False})
+    # Trigger SAS to active one DPA on channel c
+    frequency_range = {
+      'lowFrequency': 3620000000.0,
+      'highFrequency': 3630000000.0
+    }
+    self._sas_admin.TriggerDpaActivation({'frequencyRange':frequency_range,\
+                                           'dpaId':'east_dpa4'})
+    time.sleep(300)
+    # Load and register CBSD
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    # Change device location to be inside DPA neighborhood
+    device_a['installationParam']['latitude'] = 30.71570
+    device_a['installationParam']['longitude'] = -88.09350
+    cbsd_ids = self.assertRegistered([device_a]) 
+    # Create and send spectrumInquiry with the same frequency range as DPA Channel
+    spectrum_inquiry_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'spectrum_inquiry_0.json')))
+    spectrum_inquiry_0['cbsdId'] = cbsd_ids[0]
+    spectrum_inquiry_0['inquiredSpectrum'][0]['lowFrequency'] = \
+                frequency_range['lowFrequency']
+    spectrum_inquiry_0['inquiredSpectrum'][0]['highFrequency'] = \
+                frequency_range['highFrequency']
+    request = {'spectrumInquiryRequest': [spectrum_inquiry_0]}
+    response = self._sas.SpectrumInquiry(request)['spectrumInquiryResponse'][0] 
+    # Check spectrum inquiry response
+    self.assertEqual(response['cbsdId'], cbsd_ids[0])
+    self.assertTrue('availableChannel' in response)
+    self.assertEqual(response['response']['responseCode'], 0)
+    # Verify the low & high frequency are same as requested
+    self.assertEqual(response['availableChannel'][0]['frequencyRange']\
```

Comment:
Just a thought... if a SAS returned the channel availability in terms of 1 MHz or 5 MHz chunks, it would fail this test case despite having valid output.

Reply:
it is a good point! so we should treat the response with any number of channels.
i.e we should verify that those channels can form a frequency range(the unions of the channels) which contains the requested range and that those channels don't have any conflict, missing or repeated channels

Reply:
Yeah, although it's a pain to do the check :/ A helper function might be the best since we need to do something similar for SIQ.3.

Technically the text only asks us to make sure that *all* frequencies are marked available, not to check for e.g. overlapping or repeated available channels. Repeated channels might be a bug but I'm not sure overlapping channels are: suppose I know a device wants to use 15 MHz, so maybe I tell it [3550, 3565], [3555, 3570], ... with the maxEirp for each so that it can make the best decision. Another "solution" would be to give [3550, 3555], ... and let the device take the min of each 15 MHz segment. But in this case the typical [3550, 3560], ... might not be as useful for the device.

All of that to say that I don't think we need to verify that they don't overlap. If you want to add that check, let's double-check with WG3 first. You can probably also skip checking for duplicates, but WG4 may have a different opinion :)

Reply:
I agree with your opinion, and I updated the test to check that the channels form the requested frequency range and that there is no missing channels.